kernel/src/main/java/org/kframework/compile/AddImplicitCounterCell.java

```diff
@@ -21,6 +21,21 @@ public AddImplicitCounterCell() {}
 
   public Sentence apply(Module m, Sentence s) {
     if (s instanceof Claim claim) {
+      Set<KVariable> anonVars = new HashSet<>();
+      VisitK visitor =
+          new VisitK() {
+            @Override
+            public void apply(KVariable var) {
+              if (var.name().startsWith("!")
+                  || var.name().startsWith("?")
+                  || var.name().startsWith("@")) anonVars.add(var);
```

Comment:
This seems like duplicated logic that exists elsewhere. Could we pull out "extract the anonymous variables under a term" as a function?

https://github.com/runtimeverification/k/blob/41dfa2474fea88b1313e8ac2b12c25dd3e722f9e/kernel/src/main/java/org/kframework/compile/ResolveAnonVar.java#L32-L37